File: Python/identify_leader.py
from typing import List


def identify_leader(members: List[int]) -> List[int]:
    leaders = []
    leaders.append(members[-1])
    for i in range(len(members)-2, -1, -1):
        if members[i] > leaders[-1]:
            leaders.append(members[i])
    return leaders


print(identify_leader([7, 6, 9, 5, 3, 78, 9, 12, 4]))

File: CPP/Extras-Problems-Solved/identify_leader.cpp
#include <iostream>
#include <vector>

using namespace std;

vector<int> identify_leader(vector<int> &members)
{
    vector<int> leaders;
    leaders.push_back(members.back());
    int max_element = leaders[0];
    for (int i = members.size() - 2; i >= 0; i -= 1)
    {
        if (members[i] > max_element)
        {
            leaders.push_back(members[i]);
            max_element = members[i];
        }
    }
    return leaders;
}

int main()
{
    vector<int> members = {7,6,9,5,3,78,9,12,4};
    vector<int> leaders = identify_leader(members);
    cout << "The leaders are : ";
    for (const auto &num : leaders)
    {
        cout << num << " ";
    }
    return 0;
}